azerothcore/azerothcore-wotlk
# refactor(Core/Unit): PC&NPC Immunity

Co-Authored-By: Treeston https://github.com/Treeston

## Changes Proposed:
-  move UNIT_FLAG_IMMUNE_TO_PC and UNIT_FLAG_IMMUNE_TO_NPC to higher-level abstraction

New methods:
- void SetImmuneTo<All/PC/NPC>(apply, keepCombat = false);
- bool IsImmuneTo<All/PC/NPC>() const;

## Issues Addressed:

- Step for https://github.com/azerothcore/azerothcore-wotlk/issues/5985

## SOURCE:


Cherry-pick from: https://github.com/TrinityCore/TrinityCore/commit/74af880217fc567c9949d7557408718af889402f

## Tests Performed:

None

## How to Test the Changes:
- Choose a boss modified in scripts files.
- Check if everythings is good with the combat


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
